### ISSUE TITLE
Better effect casting

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#9416]
When you disagree with the effect system, we only support using a sledgehammer (`unchecked_cast (exp as _ \ target effect`). Often when you want to modify the effect of an expression you either want to remove some set of effects (e.g. #9308 or all the interop code that is non-IO) (often you want to add, but that is safe via `checked_ecast`). Using `unchecked_cast` requires you to not only give the addition/subtraction but also repeat the current effect, which is very error prone, especially if refactorings happen to the code inside the cast. I think we should have a new cast:

```
unsafe_remove <effect> { exp }
```

this would also be a safer version of `unsafe`, since `unsafe` is primarily intended to just remove IO even though it actually removes everything.

This form allows sensible programs of unchecked patterns
```
// Since Sys and IO only influences performance, it is unobservable outside this scope
unsafe_remove {Sys, IO} {
    let x = ...
    let t = Environment.handle(Environments.thread)();
    parOp(t, ...)
}
```

**Comment by JonathanStarup** [comment]
A technicality is that an effect followed by a type is ambiguous to parse (e.g. `unsafe_remove Crash + ef -a + 2`) so we must require the curly around the expression

**Comment by magnus-madsen** [comment]
I agree. We just need to bikeshed syntax. Would you implement this using difference at the type-level?

**Comment by JonathanStarup** [comment]
I would allow anything. I like the syntax as shown (don't care about the keyword itself much), you say `unsafe_remove ef { exp }` and the resulting expression has effect `exp.eff - ef`

**Comment by magnus-madsen** [comment]
Just need to bikeshed syntax.

**Comment by magnus-madsen** [comment]
How about you come up with five suggestions?

**Comment by mlutze** [comment]
This is implemented, right?

**Comment by JonathanStarup** [comment]
si, `unsafely <effect> run <exp>`
